aptos-move/framework/aptos-stdlib/sources/cryptography/bls12381.move

```diff
@@ -330,369 +400,123 @@ module aptos_std::bls12381 {
+    native fun sign_internal(sk: vector<u8>, msg: vector<u8>): vector<u8>;
+
+    #[test_only]
+    /// Generates a proof-of-possession from a secret key.
+    native fun generate_proof_of_possession_internal(sk: vector<u8>): vector<u8>;
 
-    /// Random signature generated by running `cargo test -- bls12381_sample_signature --nocapture --include-ignored` in `crates/aptos-crypto`.
-    /// The associated SK is 07416693b6b32c84abe45578728e2379f525729e5b94762435a31e65ecc728da.
-    const RANDOM_PK: vector<u8> = x"8a53e7ae5270e3e765cd8a4032c2e77c6f7e87a44ebb85bf28a4d7865565698f975346714262f9e47c6f3e0d5d951660";
 
     //
     // Tests
     //
 
     #[test_only]
-    fun get_random_aggsig(): AggrOrMultiSignature {
-        assert!(signature_subgroup_check_internal(RANDOM_SIGNATURE), 1);
-
-        AggrOrMultiSignature { bytes: RANDOM_SIGNATURE }
+    public fun maul_first_byte(bytes: &mut vector<u8>) {
+        let first_sig_byte = std::vector::borrow_mut(bytes, 0);
+        *first_sig_byte = *first_sig_byte ^ 0xff;
     }
 
     #[test_only]
-    fun get_random_pk_with_pop(): PublicKeyWithPoP {
-        assert!(validate_pubkey_internal(RANDOM_PK), 1);
-
-        PublicKeyWithPoP {
-            bytes: RANDOM_PK
+    public fun generate_proof_of_possession(sk: &SecretKey): ProofOfPossession {
+        ProofOfPossession {
+            bytes: generate_proof_of_possession_internal(sk.bytes)
         }
     }
 
     #[test]
-    fun test_pubkey_validation() {
-        // test low order points (in group for PK)
-        assert!(option::is_none(&public_key_from_bytes(x"ae3cd9403b69c20a0d455fd860e977fe6ee7140a7f091f26c860f2caccd3e0a7a7365798ac10df776675b3a67db8faa0")), 1);
-        assert!(option::is_none(&public_key_from_bytes(x"928d4862a40439a67fd76a9c7560e2ff159e770dcf688ff7b2dd165792541c88ee76c82eb77dd6e9e72c89cbf1a56a68")), 1);
-        assert!(option::is_some(&public_key_from_bytes(x"b3e4921277221e01ed71284be5e3045292b26c7f465a6fcdba53ee47edd39ec5160da3b229a73c75671024dcb36de091")), 1);
-    }
+    fun test_gen_sign_verify_normal_signature_or_signature_share() {
+        let (sk, pk) = generate_keys();
+        let pk_unvalidated = public_key_with_pop_to_normal(&pk);
 
-    #[test]
-    #[expected_failure(abort_code = 65537)]
-    fun test_empty_pubkey_aggregation() {
-        // First, make sure if no inputs are given, the function returns None
-        // assert!(aggregate_pop_verified_pubkeys(vector::empty()) == option::none(), 1);
-        aggregate_pubkeys(std::vector::empty());
+        let msg = b"hello world";
+        let sig = sign_arbitrary_bytes(&sk, msg);
+        assert!(verify_normal_signature(&sig, &pk_unvalidated, msg), 1);
+        assert!(verify_signature_share(&sig, &pk, msg), 1);
+
+        maul_first_byte(&mut sig.bytes);
+        assert!(!verify_normal_signature(&sig, &pk_unvalidated, msg), 1);
+        assert!(!verify_signature_share(&sig, &pk, msg), 1);
     }
 
     #[test]
-    fun test_pubkey_aggregation() {
```

Comment:
It's confusing when you rename previous test functions because I can't tell if the new tests are supposed to replace a particular old test.

Can you please:
1. Keep the old test names for now. If they are inadequately-named, add a TODO above them to rename them later.
   * This will allow me to check that you've maintained the same tests.
3. Feel free to add any new tests you see fit.

Reply:
If a test should be completely removed, for now just delete its contents and leave the function with an empty body, so I can see the diff easily. Then, we can fully remove it.